File: egress_hub/code/cache_order_book_updater/AppConfig/AppConfig.cpp
#include "AppConfig.h"
#include "Utils.h"

std::string AppConfig::loggerPatern = "%H:%M:%S.%e [%l] cache_order_book_updater.%n -- %v";

std::string AppConfig::hostname = Utils::getHostname();
std::string AppConfig::wild_card_port_channel = "aeron:udp?endpoint=" + AppConfig::hostname + ":0";

/* Cache Order Book Config */
int AppConfig::serviceId = 4;

/*Aeron*/
std::string AppConfig::aeronDir = Utils::getAeronDir();

/* Local ME HUB Config */
std::string AppConfig::hubInnerSignalChannel = "aeron:ipc";
int AppConfig::hubInnerSignalChannelStreamId = 1001;

std::string AppConfig::hubInnerResponseChannel = "aeron:ipc";
int AppConfig::hubInnerResponseChannelStreamId = 1002;

std::string AppConfig::memMappedMessageFile = Utils::getMemMappedMessageFile();

/*Cache Order Book*/
std::string AppConfig::cacheOrderBookUri = "tcp://redis_node";

File: egress_hub/code/hub_cluster_pusher/src/main/java/digital/exchange/egress/hub/cluster/pusher/utils/AppConfig.java
package digital.exchange.egress.hub.cluster.pusher.utils;

import io.aeron.samples.cluster.ClusterConfig;

public class AppConfig
{
    /* Common Config */
    public static final String hostname = Utils.getHostname();
    public static final String aeronWildcardPortChannel = "aeron:udp?endpoint=" + hostname + ":0";

    public static final String aeronDirName = Utils.getAeronDir();

    /* Me Egress Poller Config */
    public static final int serviceId = 2;

    /* Hub Cluster Config */
    public static final int nodeId = Utils.getClusterNode();
    public static final int portBase = Utils.getBasePort();
    public static final String hosts = Utils.getClusterAddresses();

    public static final String aeronArchiveClusterDirName = Utils.getAeronArchiveClusterDir();
    public static final String aeronClusterDirName = Utils.getAeronClusterDir();

    public static final String clusterIngressChannel = "aeron:udp";
    public static final String clusterClientEgressChannel = aeronWildcardPortChannel;

    public static final int aeronArchiveClusterControlChannelPort = portBase + ClusterConfig.ARCHIVE_CONTROL_PORT_OFFSET;
    public static final String aeronArchiveClusterControlChannel =
            "aeron:udp?endpoint=" + hostname + ":" + aeronArchiveClusterControlChannelPort;

    public static final String aeronArchiveClusterLocalControlChannel = "aeron:ipc?term-length=64k";
    public static final int aeronArchiveClusterLocalControlStreamID = 1;

    public static final String aeronArchiveClusterLocalResponseChannel = "aeron:ipc?term-length=64k";
    public static final int aeronArchiveClusterLocalResponseStreamID = 2;

    /* Cluster Archive Adapter Config */
    public static final String clusterArchiveAdapterMeReplayChannel = "aeron:udp?endpoint=" + hostname + ":11001";
    public static final int clusterArchiveAdapterMeReplayChannelStreamId = 100;

    /* Local ME HUB Config */
    public static final String hubInnerSignalChannel = "aeron:ipc";
    public static final int hubInnerSignalChannelStreamId = 1001;

    public static final String hubInnerResponseChannel = "aeron:ipc";
    public static final int hubInnerResponseChannelStreamId = 1002;
}
